src/Microsoft.DocAsCode.Plugins/DependencyType.cs

```diff
@@ -24,5 +25,27 @@ public class DependencyType
         /// the build phase that this type of dependency could have an effect on. [TO-DO]: remove the nullable ? when old value is overwritten
         /// </summary>
         public BuildPhase? Phase { get; set; }
+
+        /// <summary>
+        /// the transitivity of the dependency. [TO-DO]: remove the nullable ? when old value is overwritten
+        /// </summary>
+        public Transitivity? Transitivity { get; set; }
+
+        public bool CouldTransit(DependencyType other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+            if (Transitivity == Plugins.Transitivity.All)
```

Comment:
rename type to DependencyTransitivity?

---
In reply to: [94542184](https://github.com/dotnet/docfx/pull/1112#discussion_r94542184) [](ancestors = 94542184)